pkg/controller/restic_repository_controller.go

```diff
@@ -18,184 +18,155 @@ package controller
-	resticRepositoryClient velerov1client.ResticRepositoriesGetter,
-	kbClient client.Client,
-	repositoryManager restic.RepositoryManager,
-	defaultMaintenanceFrequency time.Duration,
-) Interface {
-	c := &resticRepositoryController{
-		genericController:           newGenericController(ResticRepo, logger),
-		resticRepositoryClient:      resticRepositoryClient,
-		resticRepositoryLister:      resticRepositoryInformer.Lister(),
-		kbClient:                    kbClient,
-		repositoryManager:           repositoryManager,
-		defaultMaintenanceFrequency: defaultMaintenanceFrequency,
-
-		clock: &clock.RealClock{},
+func NewResticRepoConciler(namespace string, logger logrus.FieldLogger, client client.Client,
+	defaultMaintenanceFrequency time.Duration, repositoryManager restic.RepositoryManager) *resticRepoConciler {
+	c := &resticRepoConciler{
+		client,
+		namespace,
+		logger,
+		clock.RealClock{},
+		defaultMaintenanceFrequency,
+		repositoryManager,
 	}
 
 	if c.defaultMaintenanceFrequency <= 0 {
 		logger.Infof("Invalid default restic maintenance frequency, setting to %v", restic.DefaultMaintenanceFrequency)
 		c.defaultMaintenanceFrequency = restic.DefaultMaintenanceFrequency
 	}
 
-	c.syncHandler = c.processQueueItem
-
-	resticRepositoryInformer.Informer().AddEventHandler(
-		cache.ResourceEventHandlerFuncs{
-			AddFunc: c.enqueue,
-		},
-	)
-
-	c.resyncPeriod = 5 * time.Minute
-	c.resyncFunc = c.enqueueAllRepositories
-
 	return c
 }
 
-// enqueueAllRepositories lists all restic repositories from cache and enqueues all
-// of them so we can check each one for maintenance.
-func (c *resticRepositoryController) enqueueAllRepositories() {
-	c.logger.Debug("resticRepositoryController.enqueueAllRepositories")
-
-	repos, err := c.resticRepositoryLister.List(labels.Everything())
-	if err != nil {
-		c.logger.WithError(errors.WithStack(err)).Error("error listing restic repositories")
-		return
-	}
-
-	for _, repo := range repos {
-		c.enqueue(repo)
-	}
+func (r *resticRepoConciler) SetupWithManager(mgr ctrl.Manager) error {
+	s := kube.NewPeriodicalEnqueueSource(r.logger, mgr.GetClient(), &velerov1api.ResticRepositoryList{}, repoSyncPeriod)
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&velerov1api.ResticRepository{}).
+		Watches(s, nil).
+		Complete(r)
 }
 
-func (c *resticRepositoryController) processQueueItem(key string) error {
-	log := c.logger.WithField("key", key)
-	log.Debug("Running processQueueItem")
-
-	ns, name, err := cache.SplitMetaNamespaceKey(key)
-	if err != nil {
-		log.WithError(errors.WithStack(err)).Error("error splitting queue key")
-		return nil
+func (r *resticRepoConciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.logger.WithField("resticRepo", req.String())
+	resticRepo := &velerov1api.ResticRepository{}
+	if err := r.Get(ctx, req.NamespacedName, resticRepo); err != nil {
+		log.WithError(err).Error("error getting restic repository")
```

Comment:
Should check whether the error is not found erorr or not

Reply:
I'll handle it